akka-http/src/main/scala/akka/http/scaladsl/coding/NoCoding.scala

```diff
@@ -17,9 +17,7 @@ import headers.HttpEncodings
 object NoCoding extends Coder with StreamDecoder {
   val encoding = HttpEncodings.identity
 
-  override def encode[T <: HttpMessage](message: T)(implicit mapper: DataMapper[T]): T#Self = message.self
   override def encodeData[T](t: T)(implicit mapper: DataMapper[T]): T = t
-  override def decode[T <: HttpMessage](message: T)(implicit mapper: DataMapper[T]): T#Self = message.self
   override def decodeData[T](t: T)(implicit mapper: DataMapper[T]): T = t
 
   val messageFilter: HttpMessage ⇒ Boolean = _ ⇒ false
```

Comment:
Hmm, I guess it would be good but otherwise we could also say that "everything that worked on 10.0.11 without deprecations will be source and binary compatible on 10.1.0".

Reply:
Yeah I guess we can say that. Probably ok to try this more aggressive scheme in HTTP's RC now and we'll see if there's panic :P

Reply:
Agreed.

Reply:
I added https://github.com/akka/akka-http/issues/1706 to have another look before releasing 10.1.0 proper.

Reply:
Ok, good!